src/main/java/teammates/ui/template/CourseEditSectionRow.java

```diff
@@ -135,19 +137,42 @@ public ElementTag getToggleSessionLevelInSectionButton() {
         
         return permissionInputGroup;
     }
-
+    
+    /**
+     * Creates a row of selections elements (checkboxes), each representing a single section
+     *   and automatically checks a single checkbox if special privileges have been assigned to
+     *   the section it corresponds to.
+     *   
+     * @param sectionNames the list of sections in the course
+     * @param sectionIndex the index of the section which will be checked for special privileges
+     * @return             a list of checkboxes, separated into rows and columns
+     */
     private List<List<ElementTag>> createSpecialSectionsForSectionRow(List<String> sectionNames, 
                                                                       int sectionIndex) {
         List<List<ElementTag>> specialSections = new ArrayList<List<ElementTag>>();
+        // i represents the row (horizontal alignment) of the checkbox
         for (int i = 0; i < sectionNames.size(); i += 3) {
             List<ElementTag> specialSectionGroup = new ArrayList<ElementTag>();
+            
+            // j represents the column (vertical alignment) of the checkbox
             for (int j = 0; j < 3 && i + j < sectionNames.size(); j++) {
+                int positionOfNewSection = i + j;
                 String name = Const.ParamsNames.INSTRUCTOR_SECTION_GROUP + sectionIndex
-                              + Const.ParamsNames.INSTRUCTOR_SECTION + (i + j);
-                ElementTag checkbox = createCheckBox(sectionNames.get(i + j), name, sectionNames.get(i + j),
-                                                     i + j == sectionIndex);
-                specialSectionGroup.add(checkbox);
+                              + Const.ParamsNames.INSTRUCTOR_SECTION + positionOfNewSection;
+                
+                if (isSectionSpecial()) {
+                    boolean isPositionMatchedWithSection = positionOfNewSection == sectionIndex;
+                    ElementTag checkbox = createCheckBox(sectionNames.get(positionOfNewSection), name,
+                                                         sectionNames.get(positionOfNewSection),
+                                                         isPositionMatchedWithSection);
+                    specialSectionGroup.add(checkbox);
```

Comment:
Since <code>specialSectionGroup.add(checkbox);</code> is duplicated in both branches, we can move it below both branches.